test/simpleSwap.js

```diff
@@ -0,0 +1,870 @@
+  const { owner, beneficiary, serial, amount, timeout } = cheque;
+  const { sig: ownerSig } = await signCheque(swap, owner, cheque); 
+  return swap.submitChequeBeneficiary(serial, amount, timeout, ownerSig, {
+    from: sender
+  });
+}
+
+async function submitChequeOwner(swap, cheque, sender) {
+    cheque = { timeout: epoch, ...cheque };
+    const { owner, beneficiary, serial, amount, timeout } = cheque;
+    const { sig: beneficiarySig } = await signCheque(swap, beneficiary, cheque);
+    return swap.submitChequeOwner(beneficiary, serial, amount, timeout, beneficiarySig, {
+      from: sender
+    });
+  }
+
+  async function submitCheque(swap, cheque, sender) {
+    cheque = { timeout: epoch, ...cheque };
+    const { owner, beneficiary, serial, amount, timeout } = cheque;
+    const { sig: ownerSig } = await signCheque(swap, owner, cheque);
+    const { sig: beneficiarySig } = await signCheque(swap, beneficiary, cheque)
+    return swap.submitCheque(beneficiary, serial, amount, timeout, ownerSig, beneficiarySig, {
+      from: sender
+    })
+  }
+
+const simpleSwapTests = (accounts, Swap) => {
+  const [owner, bob, alice] = accounts;
+
+  async function prepareSwap(prefilledAmount = 1000) {
+    const swap = await Swap.new(owner);
+    await swap.send(prefilledAmount, { from: owner });
+    return { swap, prefilledAmount: new BN(prefilledAmount) };
+  }
+
+  it("should accept deposits", async () => {
+    const { swap } = await prepareSwap(0);
+    const amount = new BN(1000);
+    const { logs } = await swap.send(amount, { from: owner });
+
+    expectEvent.inLogs(logs, "Deposit", {
+      depositor: owner,
+      amount
+    });
+
+    (await balance.current(swap.address)).should.bignumber.equal(amount);
+  });
+
+  it("should not accept a cheque with serial 0", async () => {
+    const sender = bob
+    const { swap, prefilledAmount } = await prepareSwap();
+    await shouldFail.reverting(
+      submitChequeBeneficiary(swap, {
+        owner,
+        beneficiary: bob,
+        serial: new BN(0),
+        amount: prefilledAmount
+      }, sender)
+    );
+  });
+
+  it("should accept a valid first cheque by owner", async () => {
+    const { swap, prefilledAmount } = await prepareSwap();
+
+    const cheque = {
+      owner,
+      beneficiary: bob,
+      serial: new BN(1),
+      amount: prefilledAmount,
+    };
+
+    const sender = cheque.owner
+
+    var { logs } = await submitChequeOwner(swap, cheque, sender);
+
+    expectEvent.inLogs(logs, "ChequeSubmitted", {
+      amount: cheque.amount,
+      beneficiary: cheque.beneficiary,
+      serial: cheque.serial
+    });
+
+    var chequeInfo = await swap.cheques(cheque.beneficiary);
+
+    chequeInfo.serial.should.bignumber.equal(cheque.serial);
+    chequeInfo.amount.should.bignumber.equal(cheque.amount);
+    chequeInfo.timeout.should.bignumber.gte(
+      (await time.latest()).addn(1 * epoch - 1)
+    );
+
+    // await time.increase(1 * epoch);
```

Comment:
Why is this commented out?

Reply:
It is part of the "should accept a valid first cheque by owner" test. In my opinon, checking wether the cheque can be checked out should not be part of this test. The test "should allow cheque payout after timeout" ensures this already.